frontend/src/design/components/layout/DefaultNavbar.js

```diff
@@ -17,9 +26,55 @@ const useStyles = makeStyles((theme) => ({
 
 export const DefaultNavbar = ({ openDrawer, onOpenDrawerChange }) => {
   const classes = useStyles();
+  const [isMaintenance, setMaintenanceFlag] = useState(false);
+  const dispatch = useDispatch();
+  const client = useClient();
+
+  useEffect(async () => {
+    if (client) {
+      const response = await client.query(getMaintenanceStatus());
+      if (
+        !response.errors &&
+        response.data.getMaintenanceWindowStatus !== null
+      ) {
+        if (
+          response.data.getMaintenanceWindowStatus.status === ACTIVE_STATUS ||
+          response.data.getMaintenanceWindowStatus.status === PENDING_STATUS
+        ) {
+          setMaintenanceFlag(true);
+        }
+      } else {
+        const error = response.errors
+          ? response.errors[0].message
+          : 'Could not fetch status of maintenance window';
+        dispatch({ type: SET_ERROR, error });
+      }
+    }
+  }, [client]);
```

Comment:
Can we use a try/catch block here like:

<img width="679" alt="Screenshot 2024-05-03 at 4 52 12 PM" src="https://github.com/data-dot-all/dataall/assets/97474536/f8ffbedf-62e1-4cd3-b5cc-6bd843487e72">

`
useEffect(() => {
    const fetchMaintenanceStatus = async () => {
        try {
            if (client) {
                const response = await client.query(getMaintenanceStatus());
                const maintenanceStatus = response?.data?.getMaintenanceWindowStatus;
                
                if (maintenanceStatus && [ACTIVE_STATUS, PENDING_STATUS].includes(maintenanceStatus.status)) {
                    setMaintenanceFlag(true);
                }
            }
        } catch (error) {
            const errorMessage = error?.response?.data?.errors?.[0]?.message || 'Could not fetch status of maintenance window';
            dispatch({ type: SET_ERROR, error: errorMessage });
        }
    };

    fetchMaintenanceStatus();
}, [client]);

`

Reply:
Thanks for pointing out that there is no try catch. I will decompose this into a function and call it in useEffect. I will use the `.catch` method to catch any unexpected exceptions. Similar to how it is done in `AuthGuard.js` -> `useEffect()`